Cabal/src/Distribution/Simple/GHC/BuildOrRepl.hs

```diff
@@ -97,15 +95,9 @@ buildOrReplLib mReplFlags verbosity numJobs pkg_descr lbi lib clbi = do
   -- Determine if program coverage should be enabled and if so, what
   -- '-hpcdir' should be.
   let isCoverageEnabled = libCoverage lbi
-      -- TODO: Historically HPC files have been put into a directory which
-      -- has the package name.  I'm going to avoid changing this for
-      -- now, but it would probably be better for this to be the
-      -- component ID instead...
-      pkg_name = prettyShow (PD.package pkg_descr)
-      distPref = fromFlag $ configDistPref $ configFlags lbi
       hpcdir way
         | forRepl = mempty -- HPC is not supported in ghci
-        | isCoverageEnabled = toFlag $ Hpc.mixDir distPref way pkg_name
+        | isCoverageEnabled = toFlag $ Hpc.mixDir (libTargetDir </> extraCompilationArtifacts) way
```

Comment:
and this is `targetDir` rather than `buildDir` or `tmpDir`.

Reply:
Indeed, but this `libTargetDir` is where ultimately we should be placing the extra compilation artifacts, as it refers to the library build dir... target. In fact, neither `buildDir` nor `tmpDir` are in scope there...

Reply:
There are a few naming inconsistencies amongst these functions